### PR TITLE
feat: Add BrighterScript plugin for context param diagnostics

### PR DESCRIPTION
**Pull request by chrisdp**
## Summary

- Adds a BrighterScript plugin shipped as `@rokucommunity/promises/plugin` that catches common promise mistakes at compile time
- Auto-detects ropm alias by scanning `roku_modules/` — no config needed for most projects; supports explicit alias override and multiple aliases

## Diagnostics

| Code | Severity | Description |
|------|----------|-------------|
| PRMS1001 | Warning | Context argument passed but the inline callback has no parameter to receive it |
| PRMS1002 | Warning | Chain result returned without `.toPromise()` |
| PRMS1003 | Error | Callback declares more parameters than will be supplied — extra params receive no value, crashing at runtime for non-optional params |

All three diagnostics include quick-fix code actions, with fix-all variants when multiple instances exist in the same file.

## IDE Features

- **Hover docs** — hovering over `.then`, `.catch`, or `.finally` in a `promises.chain()` call shows the expected callback signature, including the context parameter if one was passed to `chain()`
- **Snippet completions** — typing the callback argument of `onThen`, `onCatch`, `onFinally`, or `chain` surfaces `function(value, context)` snippets with tab-stops

## Changes

- `plugin/index.ts` — plugin implementation: AST walker, alias detection, three diagnostics, quick-fix actions, hover and completion providers
- `plugin/index.spec.ts` — 60 unit tests (Mocha + Chai) covering all diagnostics, code actions, alias detection, and edge cases
- `.eslintrc.js` / `.mocharc.cjs` — ESLint config adapted from BrighterScript; Mocha config for `plugin/**/*.spec.ts`
- `tsconfig.plugin.json` — compiles `plugin/` → `dist/plugin/` with declarations
- `package.json` — adds `build-plugin`, `lint`, `test-plugin` scripts; exports `./plugin` subpath; includes `dist/plugin` in published files
- `README.md` — new **BrighterScript Plugin** section with setup instructions, examples, and diagnostic code reference

## Test plan

- [ ] `npm run lint` — no errors
- [ ] `npm run test-plugin` — 60/60 passing
- [ ] `npm run build-plugin` — compiles cleanly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugin/index.ts [hunk 1]
```diff
@@ -0,0 +1,899 @@
...
+        const aliases = Array.isArray(options.alias) ? options.alias : [options.alias];
+        aliases.forEach(a => result.add(a));
+        return [...result];
+    }
+
+    // Auto-detect from roku_modules directory layout
+    const filesMap = (program as any).files as Record<string, unknown> | undefined;
+    for (const path of Object.keys(filesMap ?? {})) {
+        const normalized = path.replace(/\\/g, '/');
+        const m = /roku_modules\/([^/]+)\/source\/promises\.(?:brs|d\.bs)$/i.exec(normalized);
```
Comment (chrisdp) [inline plugin/index.ts:586]: need to update the path.
